include/envoy/server/filter_config.h

```diff
@@ -180,9 +180,10 @@ class FactoryContext : public virtual CommonFactoryContext {
   virtual Grpc::Context& grpcContext() PURE;
 
   /**
-   * @return ProcessContext& a reference to the process context.
+   * @return ProcessContext* a pointer to the process context. Will be null when running in
+   * validation mode.
    */
-  virtual ProcessContext& processContext() PURE;
+  virtual ProcessContext* processContext() PURE;
```

Comment:
nit: prefer an optional of a reference wrapper if that works for you.

Reply:
Sure, I'll make that change.

Reply:
Hmm, apparently absl::optional of a reference is [not allowed](https://github.com/abseil/abseil-cpp/blob/ac78ffc3bc0a8b295cab9a03817760fd460df2a1/absl/types/optional.h#L532).

Reply:
Done